src/vm/gcenv.ee.cpp

```diff
@@ -993,3 +993,134 @@ MethodTable* GCToEEInterface::GetFreeObjectMethodTable()
     assert(g_pFreeObjectMethodTable != nullptr);
     return g_pFreeObjectMethodTable;
 }
+
+// These are arbitrary, we shouldn't ever be having confrig keys or values
+// longer than these lengths.
+const size_t MaxConfigKeyLength = 255;
+const size_t MaxConfigValueLength = 255;
+
+bool GCToEEInterface::GetBooleanConfigValue(const char* key, bool* value)
+{
+    CONTRACTL {
+        NOTHROW;
+        GC_NOTRIGGER;
+    } CONTRACTL_END;
+
+    // these configuration values are given to us via startup flags.
+    if (strcmp(key, "gcServer") == 0)
+    {
+        *value = g_heap_type == GC_HEAP_SVR;
+        return true;
+    }
+
+    if (strcmp(key, "gcConcurrent") == 0)
+    {
+        *value = g_IGCconcurrent != 0;
+        return true;
+    }
+
+    if (strcmp(key, "GCRetainVM") == 0)
+    {
+        // GCRetainVM is strange in that it is derived from startup flags
+        // on unixes, but not on windows. This call preserves the current
```

Comment:
> or you mean you are not sure if g_pConfig->GetGCRetainVM is getting the value correctly?

For this comment, I meant that - the EE wasn't setting `g_IGCHoardVM` at all, regardless of the startup flags. This comment can probably be deleted now that this is no longer the case.